### PR TITLE
Subject Alternative Names in cert generation

### LINKED ISSUE #724: Deployed kt server's ssl cert has the wrong ip
The ssl server deployed from travis has a (self signed) cert for 0.0.0.0 and not its actual ip (currently 35.184.134.53), therefore the connection from the client fails (using a browser allows to ignore the issue).

In addition, it would be great to expose the ssl cert as a file so it is easy to download and pass to the client as an argument (but I am now looking for some openssl commands to automatically download it)

### PR DESCRIPTION
**Pull request by Liamsi**
Quickfix: These changes are necessary to enable clients (e.g. in #711) to communicate with the deployed server.
closes #724

### CODE REVIEW

**Comment by gdbelvin** [comment]
What's the status of this PR?

**Comment by Liamsi** [comment]
This PR includes a few lines of changes such that @AMarcedone's grpc (Java) client can communicate with the deployed server without warning about a wrong IP. It should create a cert that can be used from inside docker-composer, kubernetes, and externally by using the server's IP. 

I think the scripts need further cleanup but it would be good if this got merged so that the deployed server can be contacted not only from inside the kubernetes cluster.

File: scripts/gen_server_keys.sh [hunk 1]
```diff
@@ -18,21 +18,24 @@ COMMONNAME=""
...
     SUBJECT="${SUBJECT}/CN=${COMMONNAME}"
 fi
 
-SANEXT="[SAN]\nbasicConstraints=CA:TRUE\nsubjectAltName=DNS:localhost"
+# TODO(ismail): make the IPs configurable as well
+ALTNAMES="[alt_names]\nDNS.1=${SAN_DNS}\nDNS.2=localhost\nIP.1=0.0.0.0\nIP.2=35.184.134.53"
+SANEXT="[SAN]\nbasicConstraints=CA:TRUE\nsubjectAltName=@alt_names\n\n${ALTNAMES}"
 if [[ -n "${ADDRESS}" ]]; then
-    SANEXT="${SANEXT},IP.2:${ADDRESS}"
+    SANEXT="${SANEXT}"
```
Comment (AMarcedone) [inline scripts/gen_server_keys.sh]: The address option here is ignored, which might be confusing for users. I suggest fixing it before merging.

File: scripts/gen_server_keys.sh [hunk 2]
```diff
@@ -44,6 +47,8 @@ openssl genrsa -des3 -passout pass:x -out server.pass.key 2048
 openssl rsa -passin pass:x -in server.pass.key -out server.key
 chmod 600 server.key
 rm server.pass.key
+# The following command does not work on a Mac
```
Comment (AMarcedone) [inline scripts/gen_server_keys.sh]: It works on mine. Both files exists, do not remember doing anything specific to fix it.

**Comment by AMarcedone** [comment]
The deployed server works for me atm (so the cert is well formed and accepted).

**Comment by Liamsi** [comment]
> The deployed server works for me atm (so the cert is well formed and accepted).

Thanks for testing that and thanks for the review. I removed the comment about the script not working on Macs, made the san extension mandatory, and made the deployed servers address configurable. PTAL

**Comment by AMarcedone** [comment]
LGTM! Thanks for the changes and the PR in general.

**Pull request merged**